neurolang/probabilistic/query_resolution.py

```diff
@@ -161,6 +170,59 @@ def compute_probabilistic_solution(
     return solution
 
 
+def lift_solve_marg_query(rule, cpl, succ_solver):
+    """
+    Solve a MARG query on a CP-Logic program.
+
+    Parameters
+    ----------
+    query : Implication
+        Consequent must be of type `Condition`.
+        MARG query of the form `ans(x) :- P(x)`.
+    cpl_program : CPLogicProgram
+        CP-Logic program on which the query should be solved.
+
+    Returns
+    -------
+    ProvenanceAlgebraSet
+        Provenance set labelled with probabilities for each tuple in the result
+        set.
+
+    """
+    res_args = tuple(s for s in rule.consequent.args if isinstance(s, Symbol))
+
+    joint_antecedent = Conjunction(
+        tuple(
+            extract_logic_predicates(rule.antecedent.conditioned)
+            | extract_logic_predicates(rule.antecedent.conditioning)
+        )
+    )
+    joint_logic_variables = (
+        extract_logic_free_variables(joint_antecedent) & res_args
```

Comment:
this is trivially res_args, isn't it? as every variable in the consequent needs to be in the antecedent.